scipy/sparse/_dok.py

```diff
@@ -207,63 +310,43 @@ def __add__(self, other):
         if isscalarlike(other):
             res_dtype = upcast_scalar(self.dtype, other)
             new = self._dok_container(self.shape, dtype=res_dtype)
-            # Add this scalar to every element.
-            M, N = self.shape
-            for key in itertools.product(range(M), range(N)):
+            # Add this scalar to each element.
+            for key in itertools.product(*[range(d) for d in self.shape]):
```

Comment:
This is only for 2d, right? So we don't need to make it generic to nd shapes.

Reply:
Hard to tell from the diff, but this is in `__add__`, not the `_set_*` methods. So it needs to be 2d or 1d (or eventually maybe nd).